lib/Raven/Middleware/ContextInterfaceMiddleware.php

```diff
@@ -51,17 +58,32 @@ public function __construct(Context $context)
      */
     public function __invoke(Event $event, callable $next, ServerRequestInterface $request = null, \Exception $exception = null, array $payload = [])
     {
-        $tagsContext = isset($payload['tags_context']) ? $payload['tags_context'] : [];
-        $extraContext = isset($payload['extra_context']) ? $payload['extra_context'] : [];
-        $serverOsContext = isset($payload['server_os_context']) ? $payload['server_os_context'] : [];
-        $runtimeContext = isset($payload['runtime_context']) ? $payload['runtime_context'] : [];
-        $userContext = isset($payload['user_context']) ? $payload['user_context'] : [];
-
-        $event = $event->withTagsContext(array_merge($this->context->getTags(), $tagsContext), false)
-            ->withExtraContext(array_merge($this->context->getExtraData(), $extraContext), false)
-            ->withUserContext(array_merge($this->context->getUserData(), $userContext), false)
-            ->withServerOsContext($serverOsContext, false)
-            ->withRuntimeContext($runtimeContext, false);
+        if (isset($payload[$this->contextName . '_context'])) {
+            switch ($this->contextName) {
+                case Context::CONTEXT_USER:
+                    $event = $event->withUserContext($this->context->toArray(), false);
+                    $event = $event->withUserContext($payload[$this->contextName . '_context'], false);
```

Comment:
What happens if the payload doesn't have a `$this->contextName . '_context'` key? Shouldn't also you merge that beforehand, so you would call the `with` method only once?

Reply:
>What happens if the payload doesn't have a `$this->contextName . '_context'` key?

Simply nothing, the middleware will just call the next one in the chain. We cannot throw any exception or error because the `$payload` argument is optional

>Shouldn't also you merge that beforehand, so you would call the with method only once?

Merging the data with the current one is already done inside the `with*` methods, I don't really like to have to call `array_merge` all over the code and I don't see a real benefit from switching to

```php
$event = $event->withServerOsContext(array_merge($this->context->toArray(), $payload[$this->contextName . '_context']), false);
```

apart from not having to call the method twice (and cloning the event object twice). Also, I think it's more unreadable

Reply:
Ok understood.
Then, the only modification that I would do is moving `$payload[$this->contextName . '_context']` to a local variable before the switch, to reduce complexity and duplication.

Reply:
>Shouldn't also you merge that beforehand, so you would call the with method only once?

Rethinking about this you were right, in my original code if the `$payload` argument did not contain any context data then the context data set globally would had never been set in the event